### PR TITLE
added infinite scrolling and slight UI change to NewsVC

### PR DESCRIPTION
**Pull request by MattHuie**
https://gyazo.com/d0b29a4d90644b83495e2b1cc9d40371

### CODE REVIEW

File: StellarApp/Controllers/NewsViewController.swift [hunk 1]
```diff
@@ -86,7 +94,15 @@ extension NewsViewController: UICollectionViewDataSource {
         let thisArticle = articles[indexPath.row]
         cell.articleLabel.text = thisArticle.source.name
         cell.articleDescription.text = thisArticle.title
-        ImageHelper.fetchImageFromNetwork(urlString: thisArticle.urlToImage) { (error, data) in
+        
+        if (indexPath.row == articles.count - 1) {
```
Comment (vhart) [inline StellarApp/Controllers/NewsViewController.swift]: can you move this to `collectionView willShowCell`?

File: StellarApp/Controllers/View/NewsCollectionViewCell.swift [hunk 2]
```diff
@@ -50,9 +50,9 @@ class NewsCollectionViewCell: UICollectionViewCell {
     
     private func commonInit() {
        setupConstraints()
-        self.layer.cornerRadius = 5
+        self.layer.cornerRadius = 10
         self.layer.masksToBounds = true
-        self.backgroundColor = .lightGray
+        //self.backgroundColor = .lightGray
```
Comment (vhart) [inline StellarApp/Controllers/View/NewsCollectionViewCell.swift]: maybe we should delete this if it isn't being used anymore?

File: StellarApp/Controllers/View/NewsCollectionViewCell.swift [hunk 3]
```diff
@@ -86,7 +87,7 @@ class NewsCollectionViewCell: UICollectionViewCell {
             articleDescription.topAnchor.constraint(equalTo: articleLabel.bottomAnchor, constant: 4),
             articleDescription.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 12),
             articleDescription.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -12),
-            articleDescription.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12)
+            //articleDescription.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12)
```
Comment (vhart) [inline StellarApp/Controllers/View/NewsCollectionViewCell.swift]: same here

File: StellarApp/Controllers/NewsViewController.swift [hunk 4]
```diff
@@ -56,20 +60,24 @@ class NewsViewController: UIViewController {
...
         gradient.frame = self.newsView.newsCollectionView.bounds
         gradient.colors = [firstColor.cgColor,secondColor.cgColor]
         self.newsView.layer.insertSublayer(gradient, at: 0)
         newsView.newsCollectionView.delegate = self
     }
-    private func getArticles(keyword: String) {
-        ApiClient.getNews(query: keyword) { (error, data) in
+    private func getArticles(keyword: String, pageNumber: Int) {
+        ApiClient.getNews(query: keyword, page: pageNumber) { (error, data) in
             if let error = error {
```
Comment (vhart) [inline StellarApp/Controllers/NewsViewController.swift:67]: consider replacing with a guard clause.

File: StellarApp/Controllers/NewsViewController.swift [hunk 5]
```diff
@@ -56,20 +60,24 @@ class NewsViewController: UIViewController {
...
                 print(error.errorMessage())
             } else if let data = data {
+                if self.articles.isEmpty {
                 self.articles = data
+                } else {
```
Comment (vhart) [inline StellarApp/Controllers/NewsViewController.swift]: this is where you should batch update the collection view
right now you're reloading the whole thing

Comment (vhart) [reply StellarApp/Controllers/NewsViewController.swift]: https://developer.apple.com/documentation/uikit/uicollectionview/1618097-insertitems

Comment (vhart) [reply StellarApp/Controllers/NewsViewController.swift]: https://developer.apple.com/documentation/uikit/uicollectionview/1618045-performbatchupdates

**Review by vhart — CHANGES_REQUESTED** [review]
good so far. take a look at the links in the comments

File: StellarApp/Controllers/NewsViewController.swift (hunk 1 above)
Comment (MattHuie) [reply StellarApp/Controllers/NewsViewController.swift]: do you mean collectionview willdisplaycell?

**Pull request merged**